devito/stencilkernel.py

```diff
@@ -116,12 +119,16 @@ def __init__(self, stencils, **kwargs):
         nodes = SubstituteExpression(subs=subs).visit(nodes)
 
         # Apply the Devito Loop Engine for loop optimization and finalize instantiation
-        dle_state = transform(nodes, mode=dle)
+        dle_state = transform(nodes, mode=set_dle_mode(dle, self.compiler),
+                              compiler=self.compiler)
         body = dle_state.nodes
```

Comment:
I would put 
```
parameters = FindSymbols().visit(nodes)
parameters += [i.argument for i in dle_state.arguments]
```
in a `get_parameter(self, nodes, dle_state)` finction an replace by `parameters = kwrgs.get("parameters", get_parameter(nodes, dle_stat)`